Quotient/e2ee/sssshandler.cpp

```diff
@@ -0,0 +1,227 @@
+// SPDX-FileCopyrightText: 2023 Tobias Fella <tobias.fella@kde.org>
+// SPDX-License-Identifier: LGPL-2.0-or-later
+
+#include "sssshandler.h"
+
+#include "e2ee/cryptoutils.h"
+#include "room.h"
+#include "../logging_categories_p.h"
+#include "../qt_connection_util.h"
+
+using namespace Quotient;
+
+SSSSHandler::SSSSHandler(QObject* parent)
+    : QObject(parent)
+{}
+
+namespace Quotient {
+class QUOTIENT_API GetRoomKeysVersionJob : public BaseJob {
+public:
+    using BaseJob::makeRequestUrl;
+    static QUrl makeRequestUrl(QUrl baseUrl, const QUrl& mxcUri,
+                               QSize requestedSize);
+
+    GetRoomKeysVersionJob()
+        : BaseJob(HttpVerb::Get, {},
+              "/_matrix/client/v3/room_keys/version")
+    {}
+};
+
+class QUOTIENT_API GetRoomKeysJob : public BaseJob {
+public:
+    using BaseJob::makeRequestUrl;
+    static QUrl makeRequestUrl(QUrl baseUrl, const QUrl& mxcUri,
+                               QSize requestedSize);
+
+    GetRoomKeysJob(const QString& version)
+        : BaseJob(HttpVerb::Get, {},
+              "/_matrix/client/v3/room_keys/keys")
+    {
+        QUrlQuery query;
+        addParam<>(query, QStringLiteral("version"), version);
+        setRequestQuery(query);
+    }
+};
+} // namespace Quotient
+
+QByteArray SSSSHandler::decryptKey(const QString& name, const QByteArray& decryptionKey) const
+{
+    Q_ASSERT(m_connection);
+    const auto defaultKey = m_connection->accountData("m.secret_storage.default_key"_ls)->contentPart<QString>("key"_ls);
```

Comment:
Would be great to have it as an event type, too (and btw you don't really need to have a separate class for it, just use `DEFINE_SIMPLE_EVENT` before the first usage. 

This is unsafe, by the way - `accountData` may return `nullptr`. `Connection::accountDataContent<EventT>()` would be much more practical but it doesn't exist yet.